### PR TITLE
Fix SAR groups and Extra

### LINKED ISSUE #634: Subject Access Review in API RBAC is not considering Groups and Extra
# Expected Behaviour
The API server RBAC should include the Groups and Extra while doing a Subject Access Review

# Actual Behaviour
Subject Access Review doesn't consider the Groups and Extra.

# Additional Info
Hardcoded value and empty map. Ideally should be taken from TokenReview status.

https://github.com/tektoncd/results/blob/0aa7ce4900d905dfac265bd8542e99bda451d4c7/pkg/api/server/v1alpha2/auth/rbac.go#L122-L123

### PR DESCRIPTION
**Pull request by sayan-biswas**
Fixes #634 

# Changes
Add Groups and Extra in the Subject Access Review in RBAC processing.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Fix authorization check to include the Groups and Extras from the token access review.
```

### CODE REVIEW

**Comment by adambkaplan** [comment]
@sayan-biswas I added a release note - changes to behavior like this should always include a release note.

File: pkg/api/server/v1alpha2/auth/rbac.go [hunk 1]
```diff
@@ -166,11 +167,11 @@ func (r *RBAC) Check(ctx context.Context, namespace, resource, verb string) erro
 	return status.Error(codes.Unauthenticated, retMsg)
 }
 
-// convertExtra converts the map[string][]string to map[string]ExtraValue for Subject Access Review.
-func convertExtra(extra map[string][]string) map[string]authzv1.ExtraValue {
+// convertExtra converts the map[string][]string or authnv1.ExtraValue to map[string]ExtraValue for Subject Access Review.
+func convertExtra[T []string | authnv1.ExtraValue](extra map[string]T) map[string]authzv1.ExtraValue {
```
Comment (adambkaplan) [inline pkg/api/server/v1alpha2/auth/rbac.go:171]: ooo, go generics! Good godoc here for folks who are unfamiliar with how go generics work.

File: pkg/api/server/v1alpha2/auth/rbac_test.go [hunk 2]
```diff
@@ -39,8 +39,15 @@ import (
...
+		"authorized":         "a",
+		"unauthorized":       "b",
+		"authorized-group":   "c",
+		"unauthorized-group": "d",
+	}
+	// Map of users -> groups. The 'authorized' user has full permissions.
+	groups := map[string][]string{
+		"authorized-group":   {"authorized"},
+		"unauthorized-group": {"unauthorized"},
 	}
```
Comment (adambkaplan) [inline pkg/api/server/v1alpha2/auth/rbac_test.go:51]: Are you able to add a test for the extras, too? I don't think we have coverage for this yet.

**Review by adambkaplan — COMMENTED** [review]
/approve

Good fix here! I just have one request around the unit tests.

File: pkg/api/server/v1alpha2/auth/rbac_test.go (hunk 2 above)
Comment (sayan-biswas) [reply pkg/api/server/v1alpha2/auth/rbac_test.go]: Yes, I have added tests for extra now.

**Review by khrm — COMMENTED** [review]
/lgtm

**Pull request merged**